### PR TITLE
Update tsc-instrumented for project build

### PR DESCRIPTION
**Pull request by sandersn**
loggedIO has a weird build that never got updated for the project build system. This PR just adds a project for it in a straightforward way. It might be less efficient than the old way, but that's not a big concern for recording RWC test cases.

However, I may have done things wrong. If anybody knows tsc-instrumented, please comment.

### CODE REVIEW

File: src/instrumenter/instrumenter.ts [hunk 1]
```diff
@@ -15,36 +15,18 @@ ts.sys.startReplay("${ logFilename }");`);
...
-                    const index2 = index1 + invocationLine.length;
-                    const newContent = tscContent.substr(0, index1) + loggerContent + prepareCode + invocationLine + cleanupCode + tscContent.substr(index2) + "\r\n";
-                    fs.writeFile(tscPath, newContent, err => {
-                        if (err) throw err;
-                    });
-                });
-            });
-        });
-    });
+    const filename = fs.existsSync(bak) ? bak : tscPath;
```
Comment (sandersn) [inline src/instrumenter/instrumenter.ts:18]: this is mostly just refactoring to use fs.*sync, which is way easier to read -- the normal async versions are overkill for a simple rewrite script.

File: src/instrumenter/instrumenter.ts [hunk 2]
```diff
@@ -15,36 +15,18 @@ ts.sys.startReplay("${ logFilename }");`);
...
+    const tscContent = fs.readFileSync(filename, "utf-8");
+    fs.writeFileSync(bak, tscContent);
+    const loggerContent = fs.readFileSync(path.resolve(path.dirname(tscPath) + "/loggedIO.js"), "utf-8")
+    const invocationLine = "ts.executeCommandLine(ts.sys, ts.noop, ts.sys.args);";
```
Comment (sandersn) [inline src/instrumenter/instrumenter.ts:22]: the text to search for has changed slightly, and is the only real change in this file

File: src/loggedIO/loggedIO.ts [hunk 3]
```diff
@@ -205,9 +205,7 @@ namespace Playback {
         return log;
     }
 
-    function initWrapper(wrapper: PlaybackSystem, underlying: ts.System): void;
-    function initWrapper(wrapper: PlaybackIO, underlying: Harness.IO): void;
-    function initWrapper(wrapper: PlaybackSystem | PlaybackIO, underlying: ts.System | Harness.IO): void {
+    export function initWrapper(...[wrapper, underlying]: [PlaybackSystem, ts.System] | [PlaybackIO, Harness.IO]): void {
```
Comment (sandersn) [inline src/loggedIO/loggedIO.ts:208]: I can revert this if anybody cares, it's an egregious overload replaced with an egregious variadic parameter

**Review by weswigham — APPROVED** [review]

**Comment by sandersn** [comment]
Hmm, looks like `gulp tests` failed to build. It worked locally, so I need to figure out what differs between my machine and CI.

**Comment by sandersn** [comment]
Turns out I needed two tsconfigs, one for the main build and one for `tsc-instrumented`. Both `gulp tsc-instrumented` and `gulp tests` now build on my machine, but you need a `gulp clean` in between.

**Comment by sandersn** [comment]
GitHub actions are down but Devops passed.

**Pull request merged**